File: src/main/java/com/erp/system/auth/entity/Party.java
package com.erp.system.auth.entity;

import com.erp.system.common.entity.BaseAuditEntity;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "party")
@Getter
@Setter
public class Party extends BaseAuditEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String partyName;

    private String legalName;

    private String gstNumber;

    private String panNumber;

    private String status;

    @OneToMany(mappedBy = "party", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<PartyRole> partyRoles = new ArrayList<>();

    @OneToMany(mappedBy = "party", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Address> addresses = new ArrayList<>();

    @OneToMany(mappedBy = "party", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Contact> contacts = new ArrayList<>();
}

File: erp-frontend/src/types/party.ts
export interface Party {
  id: number;
  partyName: string;
  legalName?: string;
  gstNumber?: string;
  panNumber?: string;
  status?: string;
  partyRoles?: PartyRole[];
  addresses?: Address[];
  contacts?: Contact[];
  createdAt?: string;
  updatedAt?: string;
}

export interface PartyRole {
  id: number;
  roleType: string;
  party?: Party;
}

export interface Address {
  id?: number;
  party?: { id: number };
  addressType?: string;
  street?: string;
  city?: string;
  state?: string;
  country?: string;
  pincode?: string;
  isDefault?: boolean;
}

export interface Contact {
  id?: number;
  party?: { id: number };
  contactName?: string;
  email?: string;
  phone?: string;
  mobile?: string;
  designation?: string;
  isPrimary?: boolean;
}
